include/cpp2util.h

```diff
@@ -1629,39 +1629,44 @@ constexpr auto is( X&& ) {
 //  Types
 //
 template< typename C, typename X >
-auto is( X const& x ) -> bool {
+constexpr auto is( X const& x ) -> auto {
     if constexpr (
         std::is_same_v<C, X>
         || std::is_base_of_v<C, X>
     )
     {
-        return true;
+        return std::true_type{};
     }
     else if constexpr (
-        std::is_base_of_v<X, C>
-        || (
-            std::is_polymorphic_v<C>
-            && std::is_polymorphic_v<X>
-            )
+        std::is_polymorphic_v<C>
+        && std::is_polymorphic_v<X>
     )
     {
-        if constexpr (std::is_pointer_v<X>) {
-            return Dynamic_cast<C const*>(x) != nullptr;
-        }
-        else {
-            return Dynamic_cast<C const*>(&x) != nullptr;
-        }
+        return Dynamic_cast<C const*>(&x) != nullptr;
```

Comment:
Ah, because `is_pointer_v<X>` has to be `false` in this branch so it's dead code?

Reply:
In the following code:
```
        if constexpr (std::is_pointer_v<X>) {
             return Dynamic_cast<C const*>(x) != nullptr;
         }
```
the dead code is `return Dynamic_cast<C const*>(x) != nullptr;`

As `std::is_pointer_v<X>` will always be `false` just because to get to this `if`, we need to fulfill the following expression:
```
           (
            std::is_polymorphic_v<C>
            && std::is_polymorphic_v<X>
             )
```
That means `X` is polymorphic, and that means it is not a pointer (as a pointer does not fulfill the `is_polymorphic_v` type traits (see: https://godbolt.org/z/3j73cfWYe).